client/network/src/config.rs

```diff
@@ -254,6 +255,64 @@ pub fn parse_addr(mut addr: Multiaddr) -> Result<(PeerId, Multiaddr), ParseErr>
 	Ok((who, addr))
 }
 
+/// Splits a RpcAddress into a SocketAddr and PeerId.
+pub fn parse_str_rpc_addr(rpc_addr_str: &str) -> Result<(PeerId, std::net::SocketAddr), ParseErr> {
+	use regex::Regex;
+	
+	let rpc_addr_regex = Regex::new(r"^/([0-9]{1,3}\.){3}[0-9]{1,3}:[0-9]{1,5}/[a-zA-z0-9]{40,}$").unwrap();
+    if rpc_addr_regex.is_match(rpc_addr_str) {
+        let split = rpc_addr_str.split("/");
+        let vec: Vec<&str> = split.collect();
+		let who = PeerId::from_str(vec[2]).map_err(|_| ParseErr::InvalidPeerId)?;
+		let addr_rpc = std::net::SocketAddr::from_str(vec[1]).map_err(|_| ParseErr::InvalidRpcAddr)?;
+		Ok((who, addr_rpc))
+    } else {
+		Err(ParseErr::InvalidRpcAddrWithPeerId)
+	}
+}
+
+/// Rpc address of a node with peer is.
+/// 
+/// # Format
+/// /198.51.100.19:30333/QmSk5HQbn6LhUwDiNMseVUjuRYhEtYj4aUZ6WfWoGURpdV
```

Comment:
if it's our ip address, please change to a fake one

Reply:
ok